### PR TITLE
Fix custom error type description not being defined.

### LINKED ISSUE #21347: Undefined symbol error with custom error types
**Description:**
Look at the following sample:

```ballerina
public type Detail record {
    string message;
    error cause?;
};

public const RESON_FOO = "{org/module}FooError";
public type FooError error<REASON_FOO, Detail>;

public const REASON_BAR = "{org/module}BarError";
public type BarError error<REASON_BAR, Detail>;

public type Error FooError|BarError;

public type MyRecord record {
    typedesc<Error>[] myErrorTypes = [FooError];
};
```

When compiling, the compiler throws `Undefined symbol 'FooError'` error. 
 
**Steps to reproduce:**

**Affected Versions:**
Master Branch

### PR DESCRIPTION
**Pull request by irshadnilam**
## Purpose
$title

Fixes #21347

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/record/open_record.bal [hunk 1]
```diff
@@ -473,3 +473,29 @@ function addStringToMapValue(string s) returns string {
...
+    typedesc<Error>[] myErrorTypes = [FooError];
+};
+
+function testCustomErrorTypeDescFieldOnRecord() {
+    MyRecord m = {};
+    typedesc<Error> e = m.myErrorTypes[0];
+    if(!(e is typedesc<FooError>)) {
+        error panicError = error("expected typedesc<FooError> but found: " + e.toString());
+        panic panicError;
+    }
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/record/open_record.bal]: Shall we use a simple string as the reason and move the current reason you've used as the message field of the detail?

```suggestion
    if !(e is typedesc<FooError>) {
        panic error("AssertionError", message = "expected typedesc<FooError> but found: " + e.toString());
    }
```

**Comment by irshadnilam** [comment]
> This is not just for record fields.
> 
> Even the following fails,
> 
> ```ballerina
> public const FOO_REASON = "FooError";
> 
> public type FooError error<FOO_REASON>;
> 
> public function main() {
>     typedesc<error> te = FooError;
> }
> ```
> 
> Shall we update the PR description (and the issue too), and add tests?

Yes, basically custom error typedescs were not working because of this.  The code snippet you mentioned should work with this PR.

I will update the description/issue and add some tests.

File: tests/jballerina-unit-test/src/test/resources/test-src/record/open_record.bal (hunk 1 above)
Comment (irshadnilam) [reply tests/jballerina-unit-test/src/test/resources/test-src/record/open_record.bal]: +1, fixed.

**Comment by MaryamZi** [comment]
@irshadnilam shall we also update the commit message?

**Comment by irshadnilam** [comment]
> @irshadnilam shall we also update the commit message?

Updated the commit message.

**Pull request merged**